src/Components/Aspire.Azure.Messaging.EventHubs/EventProcessorClientComponent.cs

```diff
@@ -3,58 +3,62 @@
+    protected override void BindClientOptionsToConfiguration(
+        IAzureClientBuilder<EventProcessorClient, EventProcessorClientOptions> clientBuilder,
+        IConfiguration configuration)
     {
 #pragma warning disable IDE0200 // Remove unnecessary lambda expression - needed so the ConfigBinder Source Generator works
         clientBuilder.ConfigureOptions(options => configuration.Bind(options));
 #pragma warning restore IDE0200
     }
 
-    protected override void BindSettingsToConfiguration(AzureMessagingEventHubsProcessorSettings settings, IConfiguration config)
+    protected override void BindSettingsToConfiguration(AzureMessagingEventHubsProcessorSettings settings,
+        IConfiguration config)
     {
         config.Bind(settings);
     }
 
-    protected override IAzureClientBuilder<EventProcessorClient, EventProcessorClientOptions> AddClient<TBuilder>(TBuilder azureFactoryBuilder, AzureMessagingEventHubsProcessorSettings settings,
+    protected override IAzureClientBuilder<EventProcessorClient, EventProcessorClientOptions> AddClient(
+        AzureClientFactoryBuilder azureFactoryBuilder, AzureMessagingEventHubsProcessorSettings settings,
         string connectionName, string configurationSectionName)
     {
-        return azureFactoryBuilder.RegisterClientFactory<EventProcessorClient, EventProcessorClientOptions>(
-            (options, cred) =>
+        return azureFactoryBuilder.AddClient<EventProcessorClient, EventProcessorClientOptions>(
+            (options, cred, provider) =>
             {
                 EnsureConnectionStringOrNamespaceProvided(settings, connectionName, configurationSectionName);
 
                 options.Identifier ??= GenerateClientIdentifier(settings);
 
-                var blobClient = GetBlobContainerClient(settings, cred, configurationSectionName);
+                var containerClient = GetBlobContainerClient(settings, provider, configurationSectionName);
 
                 var processor = !string.IsNullOrEmpty(settings.ConnectionString)
-                    ? new EventProcessorClient(blobClient,
-                        settings.ConsumerGroup ?? EventHubConsumerClient.DefaultConsumerGroupName, settings.ConnectionString)
-                    : new EventProcessorClient(blobClient,
+                    ? new EventProcessorClient(containerClient,
+                        settings.ConsumerGroup ?? EventHubConsumerClient.DefaultConsumerGroupName,
+                        settings.ConnectionString)
+                    : new EventProcessorClient(containerClient,
                         settings.ConsumerGroup ?? EventHubConsumerClient.DefaultConsumerGroupName, settings.Namespace,
                         settings.EventHubName, cred, options);
 
                 return processor;
-
-            }, requiresCredential: false);
```

Comment:
There is no available method that provides options, cred, provider and requiresCredential. The ultimate private call has these but there is no public combination that allows to access it. Ideally on AddClient since this is the one we wanted. I assume the expectation was that if you provide a factory with `TokenCredential` you'd want `requiresCredential: true` but here we don't always.

Reply:
I was able to successfully use this using a connection string. So I'm not sure we need to specify `requiresCredential: false`. Since this is a new component, we shouldn't be breaking anyone. But let's keep the other components doing what they are doing now. This one can use AddClient.

@tg-msft - do you know what this `requiresCredential` bool does/means exactly? Why can't I specify `requiresCredential: false` when I use the overloads with IServiceProvider?

Reply:
Yeah, this is the problem I had -- there was no public overload available that allowed specifying it when you want the service provider. I couldn't immediately see what the point of this flag even was, so I figured it was vestigial as everything seemed to work regardless.

Reply:
[requiresCredential](https://learn.microsoft.com/en-us/dotnet/api/azure.core.extensions.iazureclientfactorybuilderwithcredential.registerclientfactory?view=azure-dotnet#parameters) is a niche feature for services that allow anonymous auth (i.e., reads on a Storage account used for hosting static websites).  I believe it only controls whether or not we throw an exception when resolving a client if there was no credential provided.  https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/extensions/Microsoft.Extensions.Azure/src/Internal/ClientRegistration.cs#L46